File: CSharp/Mayope.Api/Responses/ListBranchesResponse.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

using Mayope.Api.Types;

using Newtonsoft.Json;

namespace Mayope.Api.Responses;
{
    /// <summary>
    /// A response containing a list of branches.
    /// </summary>
    public class ListBranchesResponse
    {
        /// <summary>
        /// The UUID of the tree.
        /// </summary>
        [Required]
        [JsonRequired]
        public Guid Tree { get; set; }

        /// <summary>
        /// The branches in the tree, empty if none.
        /// </summary>
        [Required]
        [JsonRequired]
        public IList<Branch> Branches { get; set; }
    }
}

File: Java/org/mayope/api/response/ListBranchesResponse.java
package org.mayope.api.response;

import java.util;

import org.mayope.api.types;

/**
 * A response containing a list of branches.
 */
public class ListBranchesResponse {
    private List<Branch> branches;

    public List<Branch> getBranches() {
        return branches;
    }

    public void setBranches(List<Branch> branches) {
        this.branches = branches;
    }
}
